areas/importer/helsinki.py

```diff
@@ -0,0 +1,73 @@
+import logging
+import urllib
+
+from django.conf import settings
+from django.contrib.gis.gdal import DataSource
+from django.db import transaction
+
+from areas.models import ContractZone
+
+from .utils import ModelSyncher
+
+logger = logging.getLogger(__name__)
+
+
+class HelsinkiImporter:
+    def import_contract_zones(self):
+        logger.info('Importing Helsinki contract zones')
+
+        data_source = self._fetch_contract_zones()
+        self._process_contract_zones(data_source)
+
+        logger.info('Helsinki contract zone import done!')
+
+    def _fetch_contract_zones(self):
+        query_params = {
+            'SERVICE': 'WFS',
+            'VERSION': '2.0.0',
+            'REQUEST': 'GetFeature',
+            'TYPENAME': 'avoindata:Vastuualue_rya_urakkarajat',
+            'SRSNAME': 'EPSG:{}'.format(settings.DEFAULT_SRID),
+            'cql_filter': "tehtavakokonaisuus='PUISTO' and status='voimassa'",
+        }
+        wfs_url = '{}?{}'.format(settings.HELSINKI_WFS_BASE_URL, urllib.parse.urlencode(query_params))
+
+        logger.debug('Fetching contract zone data from url {}'.format(wfs_url))
+        data_source = DataSource(wfs_url)
+        logger.debug('Fetched {} active contract zones'.format(len(data_source[0])))
+
+        return data_source
+
+    @transaction.atomic
+    def _process_contract_zones(self, data_source):
+        layer = data_source[0]
+        syncher = ModelSyncher(ContractZone.objects.all(), lambda x: x.origin_id, self._deactivate_event)
```

Comment:
what event are you deactivating? cannot see where it comes from

Reply:
haha wtf, it should be deactivating contract zones, no idea where that "event" comes from.

Reply:
fixed